cdc/owner.go

```diff
@@ -294,6 +294,47 @@ func (o *Owner) newChangeFeed(
 	return cf, nil
 }
 
+func (o *Owner) checkAndCleanTasksInfo(ctx context.Context) error {
+	_, details, err := o.cfRWriter.GetChangeFeeds(ctx)
+	if err != nil {
+		return err
+	}
+	cleaned := false
+	for changefeedID := range details {
+		_, err := o.cfRWriter.GetAllTaskStatus(ctx, changefeedID)
+		switch errors.Cause(err) {
+		case model.ErrDecodeFailed:
+			err := o.cfRWriter.RemoveAllTaskStatus(ctx, changefeedID)
+			if err != nil {
+				return errors.Trace(err)
+			}
+			cleaned = true
+		case nil:
+		default:
+			return errors.Trace(err)
+		}
+		_, err = o.cfRWriter.GetAllTaskPositions(ctx, changefeedID)
```

Comment:
Does `/tidb/cdc/task/position/<>/<>` have compatibility changes?
If not, the task position will be retained